rest-server/src/models/job.js

```diff
@@ -387,10 +387,18 @@ class Job {
           taskStatuses: [],
         };
         for (let task of taskRoleStatuses[taskRole].taskStatuses.taskStatusArray) {
+          const containerPorts = {};
+          for (let portStr of task.containerPorts.split(';')) {
+            if (portStr.length > 0) {
+              const port = portStr.split(':');
+              containerPorts[port[0]] = port[1];
+            }
+          }
           jobDetail.taskRoles[taskRole].taskStatuses.push({
             taskIndex: task.taskIndex,
             containerId: task.containerId,
             containerIp: task.containerIp,
+            containerPorts: containerPorts,
```

Comment:
>containerPorts: containerPorts, [](start = 12, length = 31)

This will work ?

Reply:
containerPorts: containerPorts,


---
In reply to: [183613436](https://github.com/Microsoft/pai/pull/515#discussion_r183613436) [](ancestors = 183613436)

Reply:
Where's the problem? The first is a string and the second is a variable.

Reply:
use a different name ? the name is the same. it is too easy to confuse.